### ISSUE TITLE
Check properties in array to object assignments

### ISSUE THREAD

**Issue by voku** [vimeo/psalm#4362]
https://psalm.dev/r/7ce3a71807

#### working: 
because we know the type of ```self->foo```
```php
if ($property === 'foo') {
    $self->{$property} += 1; // Cannot perform a numeric operation with a non-numeric type string
}
```

#### not working: 
we know the name + type of the property from the current key-value (because it comes from array-shape phpDoc) and the types are partly not compatible with the object
```php
$self->{$property} = $value;
}
```

#### not working: 
the non-existing property ```noop``` from the array-shape assignment is not detected (see https://phpstan.org/r/4b98ab68-37e1-46a0-ac4b-550ebbff8963)
```php
/**
 * @phpstan-return array{float_test: float, foo: 'bar', lall: string, noop: int, test: int}
 */	
```
